File: 001/cpp/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> cache;
        for (int i = 0; i < nums.size(); i++) {
            int b = nums[i];
            if (auto p = cache.find(target - b); p != cache.end()) {
                return {p->second, i};
            }
            cache[b] = i;
        }
        return {};
    }
};

File: 001/rust/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut cache = HashMap::new();
        for (j, b) in nums.iter().enumerate() {
            let j = j as i32;
            match cache.get(&(target - b)) {
                Some(i) => return vec![*i, j],
                None => (),
            }
            cache.insert(b, j);
        }
        return vec![];
    }
}
